shared/lib_battery.cpp

```diff
@@ -1857,7 +1858,7 @@ double battery_t::calculate_max_discharge_kw(double *max_current_A) {
     return _voltage->calculate_max_discharge_w(q, qmax, _thermal->T_battery(), max_current_A) / 1000.;
 }
 
-double battery_t::run(size_t lifetimeIndex, double I)
+double battery_t::run(size_t lifetimeIndex, double &I)
```

Comment:
Can double &I be changed to const double &I?

Reply:
No, it needs to pass back to the caller of `run` the changes that are made to `I`. The bug this addresses is how the battery was sometimes discharging phantom power: https://github.com/NREL/SAM/issues/240#issuecomment-581515175

The reason that was happening is that the current was too high, and the `run` function would reduce it upon checking the SOC constraints, but that information was not passed back out, so the battery itself did not discharge any current, but the battery controller thought the original `I` was used.

Ideally, this logic would be performed in the dispatch classes' `check_constraints` method, but due to the iterative looping of `check_constraints` with `run`, the current would bounce back and forth rather than settling on a number which satisfies all constraints. I didn't want to modify the `check_constraints` without having more time to do so, so prioritizing the constraints in `run` seemed like the best option for now.